core/src/main/java/org/apache/brooklyn/enricher/stock/reducer/GenericStringReducer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.enricher.stock.reducer;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+
+public abstract class GenericStringReducer<T> extends Reducer<T, String>{
+
+    @Override
+    protected Function<List<T>, String> createReducerFunction(
+            String reducerName, Map<String, ?> parameters) {
+        if (reducerName.equals("formatString")){
```

Comment:
Would be good to use `Objects.equal(reducerName, "formatString")` to avoid NPE if `reducerName` is `null`.